libraries/persistent_store/src/lib.rs

```diff
@@ -362,3 +360,25 @@ pub use self::buffer::{BufferCorruptFunction, BufferOptions, BufferStorage};
 pub use self::model::{StoreModel, StoreOperation};
 pub use self::storage::{Storage, StorageError, StorageIndex, StorageResult};
 pub use self::store::{StoreError, StoreRatio, StoreResult, StoreUpdate};
+
+/// Internal representation of natural numbers.
+///
+/// In Rust natural numbers are represented as `usize`. However, internally we represent them as
+/// `u32`. This is done to preserve semantics across different targets. This is useful when tests
+/// run with `usize = u64` while the actual target has `usize = u32`.
+///
+/// To avoid too many conversions between `usize` and `Nat` which are necessary when interfacing
+/// with Rust, `usize` is used instead of `Nat` in code meant only for tests.
+///
+/// Currently, the store only supports targets with `usize = u32`.
+type Nat = u32;
```

Comment:
Nit: I'd suggest gating this declaration by `cfg(target_pointer_width = "32")` to trigger an error if someone tries it with `usize != 64`.

Reply:
I'm not sure I understand. The whole point of introducing this type is to be able to also run the code when `usize = u64` and still preserve the overflow conditions of `usize = u32`.

Reply:
I was referring to the following piece of documentation, which is currently not enforced by code.

> Currently, the store only supports targets with `usize = u32`.

Reply:
Ok, I think I got the idea. I introduced the correct gating with a comment to make it explicit.